File: Demo/src/main/java/com/myscript/iink/demo/IInkApplication.java
// Copyright MyScript. All rights reserved.

package com.myscript.iink.demo;

import android.app.Application;

import com.myscript.certificate.MyCertificate;
import com.myscript.iink.Engine;

public class IInkApplication extends Application
{
  private static Engine engine;

  public static synchronized Engine getEngine()
  {
    if (engine == null)
    {
      engine = Engine.create(MyCertificate.getBytes());
    }

    return engine;
  }

}

File: Demo/src/main/java/com/myscript/iink/demo/IInkApplication.kt
// Copyright @ MyScript. All rights reserved.

package com.myscript.iink.demo

import android.app.Application
import com.myscript.iink.demo.di.DemoModule

class IInkApplication : Application() {

    companion object {
        lateinit var DemoModule: DemoModule
    }

    override fun onCreate() {
        super.onCreate()
        DemoModule = DemoModule(this)
    }

    override fun onTerminate() {
        DemoModule.close()
        super.onTerminate()
    }
}
